### PR TITLE
chore: only allow imports from own feature or exposed field

### PR DESCRIPTION
**Pull request by schmanu**
## What it solves
Currently we can import components / services / hooks from other features that are supposed to only be used withing the feature's package.

## How this PR fixes it
Introduces new lint rules that forbid importing from different features except for components / services / hooks that are re-exported on the top-level of the feature.

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -14,7 +14,18 @@
...
     ],
-    "no-only-tests/no-only-tests": "error"
+    "no-only-tests/no-only-tests": "error",
+    "no-restricted-imports": [
+      "error",
+      {
+        "patterns": [
+          {
+            "group": ["@/features/**/*", "!@/features/*"],
+            "message": "You can not import from other features!"
```
Comment (katspaugh) [inline .eslintrc.json:24]:
```suggestion
            "message": "You cannot import from other features"
```

**Review by katspaugh — COMMENTED** [review]
Really cool!
Is there a way to have only one place where this rule is defined?

**Comment by schmanu** [comment]
> Is there a way to have only one place where this rule is defined?

I did not find a way quickly but I also did not invest that much time so far. 

I think currently it is likely also possible to circumvent the rules by importing with a relative path e.g. if in `/features/walletconnect/test` we could probably do `import * from ../../recovery/internalRecoveryPackage`

**Comment by katspaugh** [comment]
I mean, it's OK to copy this file around as long as we don't forget to do it for new features.

**Comment by katspaugh** [comment]
Shall we close or update this PR?

**Pull request closed without merging**